src/main/java/uk/gov/justice/hmpps/prison/repository/jpa/repository/CaseNoteFilter.kt

```diff
@@ -0,0 +1,84 @@
+package uk.gov.justice.hmpps.prison.repository.jpa.repository
+
+import com.google.common.collect.ImmutableList
+import jakarta.persistence.criteria.CriteriaBuilder
+import jakarta.persistence.criteria.CriteriaQuery
+import jakarta.persistence.criteria.Predicate
+import jakarta.persistence.criteria.Root
+import org.springframework.data.jpa.domain.Specification
+import uk.gov.justice.hmpps.prison.repository.jpa.model.OffenderCaseNote
+import uk.gov.justice.hmpps.prison.service.BadRequestException
+import uk.gov.justice.hmpps.prison.service.QueryParamHelper
+import java.time.LocalDate
+
+class CaseNoteFilter(
+  val bookingId: Long,
+  private val prisonId: String? = null,
+  private val startDate: LocalDate? = null,
+  private val endDate: LocalDate? = null,
+  type: String? = null,
+  subType: String? = null,
+  typesSubTypes: List<String>? = null,
+) : Specification<OffenderCaseNote> {
+
+  private val typesAndSubTypes = if (!type.isNullOrEmpty()) {
+    if (!typesSubTypes.isNullOrEmpty()) {
+      throw BadRequestException("Both type and typesSubTypes are set, please only use one to filter.")
+    }
+    if (subType.isNullOrEmpty()) listOf(type) else listOf("$type+$subType")
```

Comment:
simplify original java by adding the `type` and `subType` parameters to the list